Open Judge System/LocalWorker/OJS.LocalWorker/OjsSubmissionProcessingStrategy.cs

```diff
@@ -167,6 +166,12 @@ protected override void ProcessTestsExecutionResult(IExecutionResult<TestResult>
             this.UpdateResults();
         }
 
+        private static void UpdateParticipantTotalScore(Participant participant, int totalScore)
+        {
+            participant.TotalScoreSnapshot = totalScore;
```

Comment:
This should be in one transaction with adding/updating the participant score entity. We could have discrepancy between the snapshot and the actual score if something fails..
I suggest integrating this logic into to the logic of the `SaveParticipantScore()` method.